app/services/curation_concerns/admin_set_service.rb

```diff
@@ -1,7 +1,11 @@
 module CurationConcerns
   class AdminSetService
-    def initialize(user)
+    attr_reader :user, :context
+    delegate :blacklight_config, :repository, to: :context
+
+    def initialize(user, context)
```

Comment:
Probably better as `def initialize(context, user)`, with the more volatile argument second.  

I think this looks better:
```ruby
AdminSetService.new(self, current_user)
```
than
```ruby
AdminSetService.new(current_user, self)
```

All search services will need a context, not all will need a user.